### DISCUSSION TITLE
DevTools() is showing error

### DISCUSSION THREAD

**Discussion by EmirU116** [cefsharp/CefSharp#4391] [General]
Hello
i'm tryingt to use code 'myvariable.DevTools().Attach()' but it shows me error on 'DevTools()' saying this 

[CS0012] The type 'Control' is defined in an assembly that is not referenced. You must add a reference to assembly 'PresentationFramework, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'.

[CS1061] 'ChromiumWebBrowser' does not contain a definition for 'GetDevTools' and no accessible extension method 'GetDevTools' accepting a first argument of type 'ChromiumWebBrowser' could be found (are you missing a using directive or an assembly reference?)

maybe good to mention but im using this on Jetbrains Rider and coding on C# console application.
im using the latest version of CefSharp

![image](https://user-images.githubusercontent.com/48198544/218226219-9e4dec8d-48b3-45e8-819d-32ed209b8503.png)

**Comment by amaitland** [comment]
What are you trying to achieve exactly?

> [CS1061] 'ChromiumWebBrowser' does not contain a definition for 'GetDevTools' and no accessible extension method 'GetDevTools' accepting a first argument of type 'ChromiumWebBrowser' could be found (are you missing a using directive or an assembly reference?)

`CefSharp` doesn't have a `GetDevTools` method. Where did you get the code you are using from exactly?

> [CS0012] The type 'Control' is defined in an assembly that is not referenced. You must add a reference to assembly 'PresentationFramework, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'.

Which `Nuget` package did you install? `PresentationFramework` is used by `WPF`.

> using this on Jetbrains Rider

I've not done any testing with `Jetbrains Rider`. 

> im using the latest version of CefSharp

Please specify the exact version, `latest` is relative.

**Reply by EmirU116** [reply]
> What are you trying to achieve exactly?

got a cool little ide of making chrome close when going in on youtube, so if youtube.com is written in search bar, it should close chrome

> CefSharp doesn't have a GetDevTools method. Where did you get the code you are using from exactly?

i asked the ChatGPT and i got a bunch of code with the GetDevTools method.

> Which Nuget package did you install? PresentationFramework is used by WPF.

i installed CefSharp.Wpf version 110.0.250

**Reply by amaitland** [reply]
If you want to open `DevTools` see https://github.com/cefsharp/CefSharp/wiki/General-Usage#devtools



> i asked the ChatGPT and i got a bunch of code with the GetDevTools method.

There's a basic set of examples at https://github.com/cefsharp/CefSharp.MinimalExample

I'd suggest you start there instead of relying on `ChatGPT`.

**Reply by EmirU116** [reply]
ok thank you for answering and help :)
